File: 807-custom-sort-string/custom-sort-string.cs
public class Solution {
    public string CustomSortString(string order, string s) {
        Dictionary <char , int> freq = new Dictionary<char , int>();

        foreach(char c in s){
            if(freq.ContainsKey(c)){ freq[c]++;}else{freq[c] = 1; }
        }

        StringBuilder sb= new StringBuilder();
        int idx = 0;
        while(idx<order.Length){
            char c = order[idx];
            if(freq.ContainsKey(c) && freq[c] > 0) {
                int count = freq[c];
                while(count>0){
                 sb.Append(c); freq[c]--; 
                 count--;
                }
                 
                 }
            idx++;
        }

       foreach (var kvp in freq){
                 int count = kvp.Value;
                    while (count > 0)
                    {
                        sb.Append(kvp.Key);
                        count--;
                    }
            }

        return sb.ToString();
    }
}

File: 807-custom-sort-string/custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        unordered_map<char,int> map;


        for(int i = 0 ;i<s.length();i++){
        
            if(map.find(s[i])!=map.end()){
                map[s[i]]++;
            }else{
                map[s[i]] = 1;
            }
        }
        string result ="";
        for(int i =0; i<order.length();i++){

                if(map.find(order[i]) != map.end()){
                    while(map[order[i]] > 0){
                        map[order[i]]--;

                        result = result + order[i];  
                    }
                }

        }

        for(auto it : map){
            if(it.second >0){
                while(map[it.first] >0){
                result += it.first;
                map[it.first]--;
                }
            }
        }

        return result;

      
        
    }
};